pkg/local_object_storage/metabase/delete.go

```diff
@@ -180,3 +196,95 @@ func delUniqueIndexes(tx *bbolt.Tx, cnr cid.ID, oID oid.ID) error {
 
 	return nil
 }
+
+// forms list of objects from addrs and their missing parts.
+// [RemovedObject.PayloadLen] is not initialized.
+func supplementRemovedObjects(tx *bbolt.Tx, addrs []oid.Address) ([]RemovedObject, error) {
+	var res []RemovedObject
+	var err error
+
+	cnrMetaBktKey := make([]byte, 1+cid.Size)
+	cnrMetaBktKey[0] = metadataPrefix
+
+	for i := range addrs {
+		cnr := addrs[i].Container()
+		if slices.ContainsFunc(addrs[:i], func(a oid.Address) bool { return a.Container() == cnr }) {
```

Comment:
This can bite you with a large number of addrs. You can instead:
 * copy `addrs` into `[]RemovedObject` first (you have to do it anyway)
 * sort the list by containers once (N*log(N))
 * rely on this sorting, subslicing the set if you need to